app/controllers/documents_controller.rb

```diff
@@ -19,6 +20,14 @@ def edit
   def update
     @document = @planning_application.documents.find(params[:id])
     if @document.update(document_params)
+      if @document.saved_change_to_attribute?(:validated)
+        # we want to track cases where the document was changed from invalid to valid, hence this statement
+        if @document.saved_changes[:validated][1] == true && @document.saved_changes[:validated][0] == false
```

Comment:
Oo, I haven't really explored ActiveRecord::AttributeMethods::Dirty much before, but I think this works well. I find this syntax a little hard to read though, not knowing what is [1] and what is [2].

I noticed that [`saved_change_to_attribute?`](https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html#method-i-saved_change_to_attribute-3F) accepts and `from` and `to` option, so would this syntax work?

`if @document.saved_change_to_attribute?(:validated, from: true, to: false)`

Reply:
it does! Should have read the documentation again, thanks for this